decidim-core/app/queries/decidim/metrics/participants_metric_manage.rb

```diff
@@ -0,0 +1,164 @@
+          cumulative_value = results[:cumulative_users].count
+          next if cumulative_value.zero?
+          quantity_value = results[:quantity_users] || 0
+          space_type, space_id = key
+          record = Decidim::Metric.find_or_initialize_by(day: @day.to_s, metric_type: @metric_name,
+                                                         participatory_space_type: space_type, participatory_space_id: space_id,
+                                                         organization: @organization)
+          record.assign_attributes(cumulative: cumulative_value, quantity: quantity_value)
+          @registry << record
+        end
+        # @registry.each(&:save!)
+        @registry
+      end
+
+      private
+
+      # Creates a Hashed structure with number of Participants grouped by
+      #
+      #  - ParticipatorySpace (type & ID)
+      def query
+        return @query if @query
+
+        @query = retrieve_participatory_spaces.each_with_object({}) do |participatory_space, grouped_participants|
+          key = [participatory_space.class.name, participatory_space.id]
+          grouped_participants[key] = { cumulative_users: [], quantity_users: [] }
+          components = retrieve_components(participatory_space)
+          components.each do |component|
+            # puts "---------- (#{component.id}) #{component.manifest_name}"
+            component_participants = send(:"retrieve_participants_for_#{component.manifest_name}", component)
+            grouped_participants[key].merge!(component_participants) { |_key, g_p, c_p| g_p | c_p }
+          end
+          #   related_object = comment.root_commentable
+          #   return grouped_comments unless related_object
+          #
+          #   group_key = generate_group_key(related_object)
+          #   grouped_comments[group_key] ||= { cumulative: 0, quantity: 0 }
+          #   grouped_comments[group_key][:cumulative] += 1
+          #   grouped_comments[group_key][:quantity] += 1 if comment.created_at >= start_time
+          #
+          #   grouped_comments
+          grouped_participants
+        end
+
+        @query
+      end
+
+      # Search for all Participatory Space manifests and then all records available
+      # Limited to ParticipatoryProcesses only
+      def retrieve_participatory_spaces
+        Decidim.participatory_space_manifests.map do |space_manifest|
+          next unless space_manifest.name == :participatory_processes # Temporal limitation
+          space_manifest.participatory_spaces.call(@organization)
+        end.flatten.compact
+      end
+
+      # Search for all components published, within a fixed list of available
+      def retrieve_components(participatory_space)
+        participatory_space.components.published.where(manifest_name: AVAILABLE_COMPONENTS)
+      end
+
+      def retrieve_participants_for_proposals(component)
+        proposals = Decidim::Proposals::Proposal.where(component: component).joins(:component)
```

Comment:
This is linking `decidim-proposals` and `decidim-core`, and shouldn't be that way